src/module_loader.rs

```diff
@@ -14,19 +14,51 @@ use std::{
     rc::Rc,
 };
 
+#[allow(unused_variables)]
+#[cfg(feature = "custom_import")]
+pub trait ImportProvider {
+    fn resolve(
+        &mut self,
+        specifier: &str,

```

Comment:
Not sure I see the vaue in mirroring the deno API here vs just having rustyloader do the deno resolution and passing the trait a ModuleSpecfier?

Can you think of a usecase where you'd need to skip resolving the base ?

Reply:
> Can you think of a usecase where you'd need to skip resolving the base ?

When relative imports are intended to be resolved to URLs that aren't http/https/file. This is something I'd really like to have for my use case in using rustyscript.